tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -99,6 +101,33 @@ var _ = Describe("odo devfile create command tests", func() {
 			expectedFiles := []string{"package.json", "package-lock.json", "README.md", devfile}
 			Expect(helper.VerifyFilesExist(newContext, expectedFiles)).To(Equal(true))
 		})
+
+		It("should successfully create the devfile component and show json output for working cluster", func() {
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(devfilePath))
+			output := helper.CmdShouldPass("odo", "create", "nodejs", "--context", newContext, "-o", "json")
+			values := gjson.GetMany(output, "kind", "metadata.name", "status.state")
+			Expect(helper.GjsonMatcher(values, []string{"Component", "nodejs", "Not Pushed"})).To(Equal(true))
+		})
+
+		It("should successfully create and push the devfile component and show json output for working cluster", func() {
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(devfilePath))
+			output := helper.CmdShouldPass("odo", "create", "nodejs", "--starter", "--context", newContext, "-o", "json", "--now")
+			expectedFiles := []string{"package.json", "package-lock.json", "README.md", devfile}
+			Expect(helper.VerifyFilesExist(newContext, expectedFiles)).To(Equal(true))
+			Expect(output).To(ContainSubstring("Pushed"))
+			Expect(output).To(ContainSubstring("nodejs"))
+			Expect(output).To(ContainSubstring("Component"))
+		})
+
+		It("should successfully create the devfile component and show json output for non connected cluster", func() {
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(devfilePath))
+			kubeconfigOld := os.Getenv("KUBECONFIG")
+			os.Setenv("KUBECONFIG", "/no/such/path")
```

Comment:
I think this can cause issues for the other test cases during parallel execution.

Reply:
how do you suggest we solve this? should add the env to the specific command?

Reply:
this is something which alot of tests already do

Reply:
I think it might be a good idea to create a serialized test case for `odo create` like https://github.com/openshift/odo/blob/d6274970b4aa295618a7450f95727c1b2273a4bf/tests/integration/devfile/debug/cmd_devfile_debug_test.go#L17

Reply:
I wrote a test wrapper which allows us to do this - PTAL https://github.com/openshift/odo/pull/4472/files#diff-ecb5f46d5762ba32631a923223434ba82ea1fada752e2c01cf00db60bc73c982